service/controller/resource/cleanupenis/delete.go

```diff
@@ -2,8 +2,101 @@ package cleanupenis
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v2/pkg/controller/context/finalizerskeptcontext"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
 )
 
 func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var values []*string
+	for _, s := range cc.Status.TenantCluster.TCCP.Subnets {
```

Comment:
what about nodepools subnets? tccp subnets will most likely have only a small portion of the ENIs as most workloads run on nodepools

Reply:
Hah, I did not think about that. Good catch. How should we go about it? 

* Do we want to run one handler that takes care of everything? That handler would then be reconciled upon the `AWSCluster` CR. 
* We we want to run multiple handlers that take separately? These handlers would then be reconciled upon their respective CRs.

Reply:
I went with the first option as a starter. Please have a look if this makes sense to you.